blueflood-core/src/main/java/com/rackspacecloud/blueflood/inputs/formats/JSONMetricsContainer.java

```diff
@@ -64,14 +65,18 @@ public JSONMetricsContainer(String tenantId, List<JSONMetric> metrics) {
         delayedMetrics = new ArrayList<Metric>();
         for (JSONMetric jsonMetric : jsonMetrics) {
             Locator locator;
+            boolean isValid;
             if (jsonMetric instanceof ScopedJSONMetric) {
                 ScopedJSONMetric scopedMetric = (ScopedJSONMetric)jsonMetric;
                 locator = Locator.createLocatorFromPathComponents(scopedMetric.getTenantId(), jsonMetric.getMetricName());
+                isValid = scopedMetric.isValid();
             } else {
                 locator = Locator.createLocatorFromPathComponents(tenantId, jsonMetric.getMetricName());
+                isValid = jsonMetric.isValid();
             }
 
-            if (jsonMetric.getMetricValue() != null) {
+            // convert and add metric only if it's valid (no validation error)
+            if ((jsonMetric.getMetricValue() != null) && isValid ) {
```

Comment:
I agree that it has to change. But if I look at the method name ``toMetrics()``, It gives an impression that all it does is convert to metrics. No one will expect it to filter invalid metrics unless they look inside.  I would have named it ``getValidMetrics()``. Its not a big deal though. You can leave the way it is.